service/qos/service_level_controller.hh

```diff
@@ -51,7 +51,7 @@ public:
         virtual future<qos::service_levels_info> get_service_levels() const = 0;
         virtual future<qos::service_levels_info> get_service_level(sstring service_level_name) const = 0;
         virtual future<> set_service_level(sstring service_level_name, qos::service_level_options slo, std::optional<service::group0_guard> guard) const = 0;
-        virtual future<> drop_service_level(sstring service_level_name, std::optional<service::group0_guard> guard) const = 0;
+        virtual future<> drop_service_level(sstring service_level_name, std::vector<mutation> remove_attribute_mutation, std::optional<service::group0_guard> guard) const = 0;
```

Comment:
I think that this approach is not very flexible. What if you wanted to have a transaction (i.e. group 0 operation) which drops a service level and then also does something else?

I propose using the following principles when organizing the code:

1. If a function does a group 0 operation, then it should do the whole transaction from start to end: generate all mutations and then commit them.
2. If a function is supposed to be used in the context of a larger transaction, it should just generate mutations and return them to the caller.

This should make it easier to keep the code composable.

Here, I think we should have a separate `get_drop_service_level_functions` function which would be of type 2., while the existing `drop_service_level` would be of type 1.

Reply:
we can also separate it to a follow up, and execute 2 transactions for now, as I'll need to prepare a lot of such split funcX -> funcX.1 + funcX.2 in auth followup (let say pr should arrive next week) maybe the ones you need will be split.

but whatever is easier, I can refactor on top of our changes as well.